packages/frontend-api/src/Model/Resolver/Advert/AdvertResolverMap.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\Advert;
+
+use Overblog\GraphQLBundle\Resolver\ResolverMap;
+use Shopsys\FrameworkBundle\Component\Image\ImageFacade;
+use Shopsys\FrameworkBundle\Model\Advert\Advert;
+
+class AdvertResolverMap extends ResolverMap
+{
+    protected const RESOLVER_ADVERT_CODE = 'AdvertCode';
+    protected const RESOLVER_ADVERT_IMAGE = 'AdvertImage';
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Image\ImageFacade
+     */
+    protected $imageFacade;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Image\ImageFacade $imageFacade
+     */
+    public function __construct(ImageFacade $imageFacade)
+    {
+        $this->imageFacade = $imageFacade;
+    }
+
+    /**
+     * @return array
+     */
+    protected function map(): array
+    {
+        return [
+            'Advert' => [
+                self::RESOLVE_TYPE => function (Advert $advert) {
+                    return $this->getResolverType($advert);
+                },
+            ],
+        ];
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Advert\Advert $advert
+     * @return string
+     */
+    protected function getResolverType(Advert $advert): string
+    {
+        if ($advert->getType() === Advert::TYPE_IMAGE) {
+            return static::RESOLVER_ADVERT_IMAGE;
+        }
+        return static::RESOLVER_ADVERT_CODE;
```

Comment:
Maybe a little overthinking, but what If someone adds a new advert type. Here it will assume the new type will be code. 

What about checking for type code and throw an exception if an unknown type occurs. What do you think?

Reply:
it is a good point, I have changed it in b0b523bd8b3d865a8f5d0964067a096e717ad24a